sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -0,0 +1,469 @@
+     * @param pipeline HttpPipeline that the HTTP requests and responses flow through.
+     */
+    CryptographyAsyncClient(JsonWebKey key, HttpPipeline pipeline) {
+        Objects.requireNonNull(key);
+        this.key = key;
+        service = RestProxy.create(CryptographyService.class, pipeline);
+        if(!Strings.isNullOrEmpty(key.kid())) {
+            unpackAndValidateId(key.kid());
+            cryptographyServiceClient = new CryptographyServiceClient(key.kid(), service);
+        }
+        initializeCryptoClients();
+    }
+
+    /**
+     * Creates a CryptographyAsyncClient that uses {@code pipeline} to service requests
+     *
+     * @param kid THe Azure Key vault key identifier to use for cryptography operations.
+     * @param pipeline HttpPipeline that the HTTP requests and responses flow through.
+     */
+    CryptographyAsyncClient(String kid, HttpPipeline pipeline) {
+        unpackAndValidateId(kid);
+        service = RestProxy.create(CryptographyService.class, pipeline);
+        cryptographyServiceClient = new CryptographyServiceClient(kid, service);
+    }
+
+    private void initializeCryptoClients() {
+        if(localKeyCryptographyClient != null) {
+            return;
+        }
+        switch(key.kty()){
+            case RSA:
+            case RSA_HSM:
+                localKeyCryptographyClient = new RsaKeyCryptographyClient(key, cryptographyServiceClient);
+                break;
+            case EC:
+            case EC_HSM:
+                localKeyCryptographyClient = new EcKeyCryptographyClient(key, cryptographyServiceClient);
+                break;
+            case OCT:
+                localKeyCryptographyClient = new SymmetricKeyCryptographyClient(key, cryptographyServiceClient);
+                break;
+            default:
+                throw new IllegalArgumentException(String.format("The Json Web Key Type: %s  is not supported.", key.kty().toString()));
+        }
+    }
+
+    /**
+     * Gets the public part of the configured key. The get key operation is applicable to all key types and it requires the {@code keys/get} permission.
+     *
+     * @throws ResourceNotFoundException when the configured key doesn't exist in the key vault.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the requested {@link Key key}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Key>> getKeyWithResponse() {
+        return withContext(context -> getKeyWithResponse(context));
+    }
+
+    /**
+     * Gets the public part of the configured key. The get key operation is applicable to all key types and it requires the {@code keys/get} permission.
+     *
+     * @throws ResourceNotFoundException when the configured key doesn't exist in the key vault.
+     * @return A {@link Response} whose {@link Response#value() value} contains the requested {@link Key key}.
```

Comment:
need to update the javadoc, this method doesn't return a Response

Reply:
updated